app/models/organization.rb

```diff
@@ -72,7 +72,17 @@ def use_task_pages_api?
   end
 
   def add_user(user)
-    OrganizationsUser.find_or_create_by!(organization: self, user: user)
+    org_user = OrganizationsUser.find_or_create_by!(organization: self, user: user)
+
+    # check if membership_requests exists for the user that is being added.
+    user.membership_requests.where(organization_id: id).each do |membership_request|
```

Comment:
We only care about cancelling the pending requests, which is requests that have a status of "assigned". Ideally. I think that there should only ever be 1 pending request for a user for a specific org (I added an index to support that idea), but I think doing .each here instead of .first or find_by is probably fine. 
```suggestion
    user.membership_requests.where(organization_id: id).assigned.each do |membership_request|
```